lib/LinkEmbedder/Link.pm

```diff
@@ -382,9 +383,13 @@ __DATA__
 <a class="le-<%= $l->type %> le-provider-<%= lc $l->provider_name %>" href="<%= $l->url %>"><%= Mojo::Util::url_unescape($l->url) %></a>
 % }
 @@ video.html.ep
-<video class="le-<%= $l->type %> le-provider-<%= lc $l->provider_name %>" height="640" width="480" preload="metadata" controls>
-% for my $s (@{$l->{sources} || []}) {
-  <source src="<%= $s->{url} %>" type="<%= $s->{type} || '' %>">
-% }
-  <p>Your browser does not support the video tag.</p>
-</video>
+@@ video.html.ep
+<div class="le-<%= $l->type %> le-provider-<%= lc $l->provider_name %>" height="640" width="480" preload="metadata" controls>
+  <video class="le-<%= $l->type %> le-provider-<%= lc $l->provider_name %>" height="640" width="480" preload="metadata" controls>
+    <source src="<%= $l->{url} %>" type="<%= $l->{mimetype} || '' %>">
+    % for my $s (@{$l->{sources} || []}) {
```

Comment:
"sources" is not defined.

Reply:
I'm struggling to understand what this was supposed to do, I'm under the impression this was not working.
I'll take another look, and remove it if not needed.

Reply:
Sorry about this one. I thought this was something you added, but I got tricked by the indentation.